packages/ember/tests/helpers/link_to_test/link_to_with_query_params_test.js

```diff
@@ -1,506 +1,567 @@
-import { Route, NoneLocation } from 'ember-routing';
-import { compile } from 'ember-template-compiler';
-import { Application } from 'ember-application';
-import { jQuery } from 'ember-views';
-import { setTemplates, setTemplate } from 'ember-glimmer';
-
-let Router, App, router, registry, container;
-
-function bootApplication() {
-  router = container.lookup('router:main');
-  run(App, 'advanceReadiness');
-}
-
-
-function shouldNotBeActive(selector) {
-  checkActive(selector, false);
-}
-
-function shouldBeActive(selector) {
-  checkActive(selector, true);
-}
-
-function checkActive(selector, active) {
-  let classList = jQuery(selector, '#qunit-fixture')[0].className;
-  equal(classList.indexOf('active') > -1, active, selector + ' active should be ' + active.toString());
-}
-
-let updateCount, replaceCount;
-
-function sharedSetup() {
-  App = Application.create({
-    name: 'App',
-    rootElement: '#qunit-fixture'
-  });
-
-  App.deferReadiness();
-
-  updateCount = replaceCount = 0;
-  App.Router.reopen({
-    location: NoneLocation.create({
-      setURL(path) {
-        updateCount++;
-        set(this, 'path', path);
-      },
-
-      replaceURL(path) {
-        replaceCount++;
-        set(this, 'path', path);
-      }
-    })
-  });
-
-  Router = App.Router;
-  registry = App.__registry__;
-  container = App.__container__;
-}
-
-function sharedTeardown() {
-  run(() => App.destroy());
-  setTemplates({});
-}
-
-QUnit.module('The {{link-to}} helper: invoking with query params', {
-  setup() {
-    run(() => {
-      sharedSetup();
-
-      App.IndexController = Controller.extend({
-        queryParams: ['foo', 'bar', 'abool'],
-        foo: '123',
-        bar: 'abc',
-        boundThing: 'OMG',
-        abool: true
-      });
-
-      App.AboutController = Controller.extend({
+import { Route, Router } from 'ember-routing';
+import { moduleFor, ApplicationTestCase, AutobootApplicationTestCase } from 'internal-test-helpers';
+
+moduleFor('The {{link-to}} helper: invoking with query params', class extends ApplicationTestCase {
+  constructor() {
+    super();
+    let indexProperties = {
+      foo: '123',
+      bar: 'abc'
+    }
+    this.add('controller:index', Controller.extend({
+      queryParams: ['foo', 'bar', 'abool'],
+      foo: indexProperties.foo,
+      bar: indexProperties.bar,
+      boundThing: 'OMG',
+      abool: true
+    }));
+    this.add('controller:about', Controller.extend({
         queryParams: ['baz', 'bat'],
         baz: 'alex',
         bat: 'borf'
-      });
-
-      registry.unregister('router:main');
-      registry.register('router:main', Router);
+    }));
+    this.indexProperties = indexProperties;
+  }
+
+  shouldNotBeActive(selector) {
+    this.checkActive(selector, false);
+  }
+
+  shouldBeActive(selector) {
+    this.checkActive(selector, true);
+  }
+
+  checkActive(selector, active) {
+    let classList = this.$(selector, '#qunit-fixture')[0].className;
```

Comment:
`this.$(selector)` will already be scoped here, no need to pass a second argument.